### PR TITLE
Sync 'kitAssembly' collection to BigQuery

### PR DESCRIPTION
**Pull request by we-ai**
This PR addresses issue [#818](https://github.com/episphere/connect/issues/818), and code cleanup.

### CODE REVIEW

File: utils/events.js [hunk 1]
```diff
@@ -2,119 +2,100 @@ const firestore = require('@google-cloud/firestore');
...
-  
-    // load() waits for the job to finish
-    console.log(`Load job ${job.id} completed.`);
-  
-    // Check the job's status for errors
-    const errors = job.status.errors;
-    if (errors && errors.length > 0) {
-        throw errors;
-    }
+const firestoreExport = async (event, context) => {
```
Comment (Gbarra9) [inline utils/events.js:7]: Are the parameters `event` and `context` for `firestoreExport` going to be added at a later time?

File: utils/events.js [hunk 2]
```diff
@@ -2,119 +2,100 @@ const firestore = require('@google-cloud/firestore');
...
 
-const firestoreExport = async (event, context) => {
-    const client = new firestore.v1.FirestoreAdminClient();
-    const gcsBucket = process.env.GCLOUD_BUCKET;
-    console.log(gcsBucket)
-    const bucket = `gs://${gcsBucket}`;
-    const projectId = process.env.GCP_PROJECT || process.env.GCLOUD_PROJECT;
-    console.log(projectId)
-    const databaseName = client.databasePath(projectId, '(default)');
+const importToBigQuery = async (gcsEvent, context) => {
```
Comment (Gbarra9) [inline utils/events.js:11]: Similar question to the parameters in `firestoreExport`. The parameter `context` is not being used in `importToBigQuery`

File: utils/events.js [hunk 3]
```diff
@@ -2,119 +2,100 @@ const firestore = require('@google-cloud/firestore');
...
+  await importCollectionsToBigQuery(gcsEvent, collectionNameArray);
+};
 
-    await client.exportDocuments({
-        name: databaseName,
-        outputUriPrefix: bucket,
-        collectionIds: collectionNameList,
-    });
+const exportNotificationsToBucket = async (event, context) => {
```
Comment (Gbarra9) [inline utils/events.js:15]: Similar question to the parameters in `firestoreExport`. The parameters `event` and `context` are not being used in `exportNotificationsToBucket`

File: utils/events.js [hunk 4]
```diff
@@ -2,119 +2,100 @@ const firestore = require('@google-cloud/firestore');
...
+  await exportCollectionsToBucket(["notifications"]);
+};
 
-    return true;
+const importNotificationsToBigquery = async (gcsEvent, context) => {
```
Comment (Gbarra9) [inline utils/events.js:19]: Similar question to the parameters in `firestoreExport`. The parameter `context` is not being used in `importNotificationsToBigquery`.

**Review by Gbarra9 — APPROVED** [review]

File: utils/events.js (hunk 4 above)
Comment (we-ai) [reply utils/events.js:19]: The 4 functions (`firestoreExport`, `importToBigQuery`, `exportNotificationsToBucket`, `importNotificationsToBigquery`) are triggered by events (Pub/Sub or Bucket events). Reference [here](https://cloud.google.com/functions/docs/writing/write-event-driven-functions#background-functions) about event-driven functions. These parameters are passed to defined handlers when these handlers are called. `event` has event data payload, and `context` has metadata about the event.

So, even though the 4 function does not currently use all of parameters (`event`, `context`), it's still reasonable to include them as they are part of the function signature for Google Cloud Functions.

Comment (Gbarra9) [reply utils/events.js:19]: Makes sense. Thanks for informing me about this.

File: utils/events.js [hunk 5]
```diff
@@ -2,119 +2,100 @@ const firestore = require('@google-cloud/firestore');
...
-    const gcsEvent = event;
-    let tableName = ''
-    if (gcsEvent.name.indexOf('.export_metadata') === -1) return true;
-    for (const collectionName  of collectionNameList) {
-        if (gcsEvent.name.includes(collectionName)) {
-            tableName = collectionName;
-            break;
-        }
-    }
+const collectionNameArray = ['participants','biospecimen', 'boxes', 'module1_v1', 'module1_v2', 'module2_v1', 'module2_v2', 'module3_v1', 'module4_v1', 'bioSurvey_v1', 'menstrualSurvey_v1', 'clinicalBioSurvey_v1','covid19Survey_v1', 'kitAssembly'];
```
Comment (jhflorey) [inline utils/events.js:5]: For issue https://github.com/episphere/connect/issues/766 i added `sendgridTracking` collection. Should we add it  into  `collectionNameArray`

Comment (we-ai) [reply utils/events.js:5]: I don't decide whether it should be synced to BQ, and I didn't receive such request.

Comment (jhflorey) [reply utils/events.js:5]: Got it. Should update later

**Review by jhflorey — APPROVED** [review]

**Review by JoeArmani — APPROVED** [review]
looks good, no comments to add

**Pull request merged**